src/Maths/sphericalPolynomial.ts

```diff
@@ -152,20 +153,31 @@ export class SphericalHarmonics {
      * @param deltaSolidAngle the delta solid angle of the light
      */
     public addLight(direction: Vector3, color: Color3, deltaSolidAngle: number): void {
-        var colorVector = new Vector3(color.r, color.g, color.b);
-        var c = colorVector.scale(deltaSolidAngle);
-
-        this.l00 = this.l00.add(c.scale(applySH3(0, direction)));
-
-        this.l1_1 = this.l1_1.add(c.scale(applySH3(1, direction)));
-        this.l10 = this.l10.add(c.scale(applySH3(2, direction)));
-        this.l11 = this.l11.add(c.scale(applySH3(3, direction)));
-
-        this.l2_2 = this.l2_2.add(c.scale(applySH3(4, direction)));
-        this.l2_1 = this.l2_1.add(c.scale(applySH3(5, direction)));
-        this.l20 = this.l20.add(c.scale(applySH3(6, direction)));
-        this.l21 = this.l21.add(c.scale(applySH3(7, direction)));
-        this.l22 = this.l22.add(c.scale(applySH3(8, direction)));
+        TmpVectors.Vector3[0].set(color.r, color.g, color.b);
+        const colorVector = TmpVectors.Vector3[0];
+        const c = TmpVectors.Vector3[1];
+        colorVector.scaleToRef(deltaSolidAngle, c);
+
+        c.scaleToRef(applySH3(0, direction), TmpVectors.Vector3[2]);
+        this.l00 = this.l00.addInPlace(TmpVectors.Vector3[2]);

```

Comment:
nit: you don't need `this.l00 =` (also you don't do it for the other parameters below, so it's more homogeneous)